File: python/leetcode/405.py
class Solution(object):
    def toHex(self, num):
        if num == 0: return "0"
        elif num < 0: num += 1 << 32
        converthex, res = ["0","1","2","3","4","5","6","7","8","9","a","b","c","d","e","f"], ""
        while num: num, res = num//16, converthex[num%16]+res
        return res

File: cpp/leetcode/405.cpp
class Solution {
public:
    string toHex(int num) {
        if(num == 0) return "0";
        string res = "";
        for(int i = 0; i < 8; i++){
            res = map[num & 15] + res; // 1111b
            num >>= 4;
        }
        auto idx = res.find_first_not_of('0');
        return res.substr(idx);
    }
private:
    const string map = "0123456789abcdef";
};